### PR TITLE
Fix SDPA mask scaling to match PyTorch semantics

### LINKED ISSUE #42694: PCC drop in SDPA,PCC=0.9444633359403863
### Describe the bug

- PCC drop observed in SDPA, PCC=`0.9444633359403863` 
- Beit's Base variant is affected by this issue - https://github.com/tenstorrent/tt-xla/issues/4314

### Repro

```
git checkout kkannan/apr20_sdpa_pcc_drop
git submodule update --init --recursive

pytest tests/ttnn/unit_tests/operations/sdpa_beit_base/test_sdpa.py -svv
```

### Log

- [apr20_sdpa_pcc_drop.log](https://github.com/user-attachments/files/26904285/apr20_sdpa_before_fix.log)

### PR DESCRIPTION
**Pull request by kamalrajkannan78**
### Ticket

- fixes https://github.com/tenstorrent/tt-metal/issues/42694

### Problem description

- `ttnn.transformer.scaled_dot_product_attention` diverges from PyTorch's F.scaled_dot_product_attention when scale != 1.0 and a user attn_mask is supplied.
- PyTorch: softmax(Q·Kᵀ * scale + mask) — scale applies to Q·Kᵀ only.
- The compute kernel folds scale into the softmax exponent as: `exp((QK + mask - max) * scale) = exp(QK*scale + mask*scale - max*scale)` ([Code evidence](https://github.com/tenstorrent/tt-metal/blob/kkannan/apr22_fix_sdpa_drop/ttnn/cpp/ttnn/operations/transformer/sdpa/device/kernels/compute/compute_common.hpp#L1867-L1925)) which silently multiplies the mask by scale too. 
- For any scale < 1, the user's mask is diluted in softmax. 

### What's changed

- Pre-multiply attn_mask by 1/scale in the host wrapper: `exp(QK*scale + (mask/scale)*scale - max*scale) = exp(QK*scale + mask - max*scale)`, restoring PyTorch semantics.               
- Guarded on attn_mask.has_value() && scale != 1.0f.
-  BEiT SDPA PCC: **0.9444 → 0.9941**. 

### Checklist

- [x] L2 Nightly - https://github.com/tenstorrent/tt-metal/actions/runs/24788047149
- [x] Sanity - https://github.com/tenstorrent/tt-metal/actions/runs/24787685958

### Logs

- [apr23_sdpa_before_fix.log](https://github.com/user-attachments/files/26995768/apr23_sdpa_before_fix.log)
- [apr23_sdpa_after_fix.log](https://github.com/user-attachments/files/26995767/apr23_sdpa_after_fix.log)

### CODE REVIEW

**Review by saiarthiraguram — APPROVED** [review]

**Comment by skrsticTT** [comment]
maybe we can do the same for `flash_mla_prefill`, it looks like it has the same bug

**Comment by kamalrajkannan78** [comment]
> maybe we can do the same for `flash_mla_prefill`, it looks like it has the same bug

Maybe we can handle this in a separate PR, since it would require re-triggering the nightly and sanity runs?

**Comment by skrsticTT** [comment]
ok, but pls open an issue for it

**Review by skrsticTT — APPROVED** [review]

**Comment by kamalrajkannan78** [comment]
> ok, but pls open an issue for it

https://github.com/tenstorrent/tt-metal/issues/43239
